applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+	/* Audio data block to input */
+	struct aobj_block block;
+
+	/* Sending module's handle */
+	struct amod_handle *tx_handle;
+
+	/* Callback for when the data has been consumed */
+	amod_response_cb response_cb;
+};
+
+/**
+ * @brief A helper structure for describing the modules in a stream as a table.
+ */
+struct amod_table {
+	/* A unique name for the module */
+	char *name;
+
+	/* A unique handle to the module */
+	struct amod_handle *handle;
+
+	/* A unique handle to the module */
+	struct amod_configuration *initial_config;
+
+	/* the modules parameters */
+	struct amod_parameters params;
+
+	/* A unique module context */
+	struct amod_context *context;
+};
+
+/**
+ * @brief  Function for opening a module.
+ *
+ * @param parameters     Pointer to the module set-up parameters
+ * @param configuration  A pointer to the modules configuration (this must also
+ *                       be passed to the #amod_set_configuration() unchanged)
+ * @param name           A unique name for this instance of the module
+ * @param handle         Pointer to the module's private handle
+ * @param context        Pointer to the private context for the module
+ *
+ * @return 0 if successful, error value
+ */
+int amod_open(struct amod_parameters *parameters, struct amod_configuration *configuration,
+	      char *name, struct amod_handle *handle, struct amod_context *context);
+
+/**
+ * @brief  Function to close an open module.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_close(struct amod_handle *handle);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules configuration to set (as
+ *                       passed to the modules #amod_query_resource())
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_set(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules current configuration
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_get(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief Function to connect two modules together.
+ *
+ * @param handle_from  The handle for the module for output
+ * @param handle_to    The handle of the module to for input, if the same as ::handle from
+ *                     the data will be put on the ::handle_from output message quere.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_connect(struct amod_handle *handle_from, struct amod_handle *handle_to);
+
+/**
+ * @brief Function to disconnect a module.
+ *
+ * @param handle             The handle for the module
+ * @param handle_disconnect  The handle of the module to disconnect
```

Comment:
Aren't both modules "equals", meaning that they "disconnect from each other"?

Reply:
Yes I see what you mean now. It they are the same the module will not send data back via it's own output FIFO. In effect it stst the modules handle->data_tx to false.